### PR TITLE
Feature/sse c base64

### LINKED ISSUE #6400: You cannot use a full binary value for sse_customer_key
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/cant-use-sse-customer-key-with-non-printable-charactersin-config-file/32667

#### What is your current rclone version (output from `rclone version`)?

tried with v1.56.2 and v1.50.2

#### What problem are you are trying to solve?

There does not seem to be any way to put a non-printable sse_customer_key in the config file -- also, that key should probably be obscured anyway.

#### How do you think rclone should be changed to solve that?

The easiest method that occurs to me is to just allow it to be obscured; since you can obscure a non-printable key and it results in printable characters that should do it. The other option I see would be to detect in some way if it is a base64 encoded value, e.g. "this is longer than 32 bytes and it's an AES256 key, so it must be b64 encoded" or supporting `base64:<b64 encoded customer key>`

Seems like it should be a simple enough change.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

### PR DESCRIPTION
**Pull request by taxilian**
#### What is the purpose of this change?

Fixes #6400 -- make it possible to provide a binary sse_customer_key for s3 SSE-C usage

#### Was the change discussed in an issue or in the forum before?

Yes, see #6400 and https://forum.rclone.org/t/cant-use-sse-customer-key-with-non-printable-charactersin-config-file/32667

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/s3/s3.go [hunk 1]
```diff
@@ -2644,6 +2654,17 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 	if opt.BucketACL == "" {
 		opt.BucketACL = opt.ACL
 	}
+	if opt.SSECustomerKeyB64 != "" && opt.SSECustomerKey != "" {
+		return nil, errors.New("s3: can't use sse_customer_key and sse_customer_key_base64 at the same time")
+	}
+	if opt.SSECustomerKeyB64 != "" && opt.SSECustomerKey == "" {
```
Comment (ncw) [inline backend/s3/s3.go:2660]: The `&& opt.SSECustomerKey == ""` is redundant due to the if above I think.

File: backend/s3/s3.go [hunk 2]
```diff
@@ -1536,7 +1536,16 @@ isn't set then "acl" is used instead.`,
 			}},
 		}, {
 			Name:     "sse_customer_key",
-			Help:     "If using SSE-C you must provide the secret encryption key used to encrypt/decrypt your data.",
+			Help:     "If using SSE-C you must provide the secret encryption key used to encrypt/decrypt your data. Alternatively you can provide sse_customer_key_base64.",
```
Comment (ncw) [inline backend/s3/s3.go:1539]: The `Help:` is used to make to make the docs and also the help for the flag.

Only the first line is used for the flag help text (try `rclone help flags sse`)

The other lines should be separated by a blank line and wrapped to 80 chars.

So I think this needs to be something like this. Note that we usually
refer to options by their flag name in the docs.

```go
			Help:     `Secret encryption key for SSE-C

If using SSE-C you must provide the secret encryption key used to
encrypt/decrypt your data. Alternatively you can provide
--sse-customer-key-base64.`,

```

File: backend/s3/s3.go [hunk 3]
```diff
@@ -1536,7 +1536,16 @@ isn't set then "acl" is used instead.`,
...
+			Provider: "AWS,Ceph,ChinaMobile,Minio",
+			Advanced: true,
+			Examples: []fs.OptionExample{{
+				Value: "",
+				Help:  "None",
+			}},
+		}, {
+			Name:     "sse_customer_key_base64",
+			Help:     "If using SSE-C you must provide the secret encryption key used to encrypt/decrypt your data in base64 format. Alternatively you can provide sse_customer_key.",
```
Comment (ncw) [inline backend/s3/s3.go:1548]: This help needs to change too.

File: backend/s3/s3.go [hunk 4]
```diff
@@ -2111,6 +2120,7 @@ type Options struct {
 	SSEKMSKeyID           string               `config:"sse_kms_key_id"`
 	SSECustomerAlgorithm  string               `config:"sse_customer_algorithm"`
 	SSECustomerKey        string               `config:"sse_customer_key"`
+	SSECustomerKeyB64     string               `config:"sse_customer_key_base64"`
```
Comment (ncw) [inline backend/s3/s3.go:2123]: I prefer these to match so `SSECustomerKeyBase64` even if it is a bit of a mouthful!

**Review by ncw — COMMENTED** [review]
Looking very good :-)

I put some comments about help text (which was wrong before your change!) and naming inline.

Thank you :-)

File: backend/s3/s3.go (hunk 1 above)
Comment (taxilian) [reply backend/s3/s3.go]: Changed -- I also made it an `else if`, which is technically irrelevant, but I feel like it makes the code more clear that those two are related and the following one is not (or not directly, anyway -- it's about making sure we have the md5, not about the key field itsself)

File: backend/s3/s3.go (hunk 4 above)
Comment (taxilian) [reply backend/s3/s3.go]: done

**Comment by taxilian** [comment]
Rewrote the commit messages to match your requested format (I think, I have a short attention span sometimes) =]

**Review by ncw — APPROVED** [review]
That looks great, thanks.

I'll merge that now :-)

**Comment by taxilian** [comment]
Not trying to jog your elbow, just wondering -- when do you expect to have a release with this in it? It looks like it just missed the last one?

**Comment by ncw** [comment]
This week is the plan!

**Pull request merged**